### PR TITLE
Alias deprecated TexRefInputIterator to TexObjInputIterator.

### PR DESCRIPTION
**Pull request by allisonvacanti**
Removes usage of deprecated CUDART texture reference APIs without
breaking CUB's API.

Other changes:
- Remove obsolete checks for CUDART_VERSION >= 5.5.
- Split `test_iterator.cu` to create a test that only handles the
  deprecated `TexRefInputIterator` API.
  - Reduces the scope of our deprecation suppressions.
  - Also removed the deprecation suppression from our CMake logic.
- Enable testing for `TexObjInputIterator` without `CUB_CDP`.
- Add a meaningful implementation for `ostream << TexObjInputIterator`.

### CODE REVIEW

File: cub/iterator/tex_ref_input_iterator.cuh [hunk 1]
```diff
@@ -225,212 +105,10 @@ typename IteratorTexRef<T>::template TexId<UNIQUE_ID>::TexRef IteratorTexRef<T>:
...
-#pragma warning(default:4996)
-#elif CUB_HOST_COMPILER == CUB_HOST_COMPILER_GCC || \
-      CUB_HOST_COMPILER == CUB_HOST_COMPILER_CLANG
-#pragma GCC diagnostic pop
-#endif
-
-};
-
-
+using TexRefInputIterator CUB_DEPRECATED = cub::TexObjInputIterator<T, OffsetT>;
```
Comment (senior-zero) [inline cub/iterator/tex_ref_input_iterator.cuh:110]: Does this positioning of `CUB_DEPRECATED` work with all supported modes? I've checked latest MSVC with C++11 and got [some issues](https://godbolt.org/z/7846v4zaM).

File: cub/iterator/tex_ref_input_iterator.cuh [hunk 2]
```diff
@@ -33,16 +33,11 @@
...
+#include <cub/iterator/tex_obj_input_iterator.cuh>
 
-#include "../thread/thread_load.cuh"
-#include "../thread/thread_store.cuh"
-#include "../util_device.cuh"
-#include "../util_debug.cuh"
-#include "../config.cuh"
-
-#if (CUDART_VERSION >= 5050) || defined(DOXYGEN_ACTIVE)  // This iterator is compatible with CUDA 5.5 and newer
+#include <iostream>
```
[the remark this answers is not in the document]
Comment (allisonvacanti) [reply cub/iterator/tex_ref_input_iterator.cuh:39]: Agreed, I'll remove the other headers.

File: cub/iterator/tex_ref_input_iterator.cuh [hunk 3]
```diff
@@ -51,128 +46,13 @@
 
 CUB_NAMESPACE_BEGIN
 
-
-/******************************************************************************
- * Static file-scope Tesla/Fermi-style texture references
- *****************************************************************************/
-
-#ifndef DOXYGEN_SHOULD_SKIP_THIS    // Do not document
```
[the remark this answers is not in the document]
Comment (allisonvacanti) [reply cub/iterator/tex_ref_input_iterator.cuh:59]: Discussed offline, this should be fine.

File: cub/iterator/tex_ref_input_iterator.cuh (hunk 2 above)
[the remark this answers is not in the document]
Comment (allisonvacanti) [reply cub/iterator/tex_ref_input_iterator.cuh]: Done.

**Review by senior-zero — APPROVED** [review]

**Pull request merged**